src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -257,58 +254,53 @@ internal static ProcessInfo[] GetProcessInfos(Predicate<int>? processIdFilter =
             ProcessInfo[] processInfos;
 
             // Start with the default buffer size.
-            int bufferSize = DefaultCachedBufferSize;
+            int bufferSize = DefaultCachedBufferSize * sizeof(long);
 
             // Get the cached buffer.
-            long[]? buffer = Interlocked.Exchange(ref CachedBuffer, null);
+            byte[] buffer = ArrayPool<byte>.Shared.Rent(bufferSize);
 
-            try
+            while (true)
             {
-                while (true)
-                {
-                    if (buffer == null)
-                    {
-                        // Allocate buffer of longs since some platforms require the buffer to be 64-bit aligned.
-                        buffer = new long[(bufferSize + 7) / 8];
-                    }
+                uint requiredSize = 0;
 
-                    uint requiredSize = 0;
-
-                    unsafe
+                unsafe
+                {
+                    // Note that the buffer will contain pointers to itself and it needs to be pinned while it is being processed
+                    // by GetProcessInfos below
+                    fixed (byte* bufferPtr = buffer)
                     {
-                        // Note that the buffer will contain pointers to itself and it needs to be pinned while it is being processed
-                        // by GetProcessInfos below
-                        fixed (long* bufferPtr = buffer)
+                        // some platforms require the buffer to be 64-bit aligned.
+                        // ArrayPool does not guarantee 64-byte alignment, so we take the address of first 64-byte aligned element
+                        // we round up the address of the pinned array to the nearest multiple of 64
+                        Debug.Assert(bufferSize > 64);
+                        byte* alignedBufferPtr = (byte*)(((nint)bufferPtr + 63) & ~63);
```

Comment:
@eiriktsarpalis the proof that the magic works:

```
static void Main(string[] args)
{
    if (GetFirstAlignedIndex(0) != 0) throw new Exception();

    for (int i = 1; i <= 64; i++)
    {
        if (GetFirstAlignedIndex(i) != 64 - i) throw new Exception();
    }

    static int GetFirstAlignedIndex(int address)
    {
        int rounded = (address + 63) & ~63;
        return rounded - address;
    }
}
```

Reply:
Are there any arch-specific concerns we should be accounting for here?

Reply:
> Are there any arch-specific concerns we should be accounting for here?

Using native int here should protect us from it